core/src/main/java/org/apache/calcite/config/CalciteSystemProperty.java

```diff
@@ -131,6 +131,15 @@
   public static final CalciteSystemProperty<Boolean> TOPDOWN_OPT =
       booleanProperty("calcite.planner.topdown.opt", false);
 
+  /**
+   * Whether to enable index-based access for struct fields.
+   *
+   * <p>Note: the feature is experimental as it relies on field order which is JVM-dependent
+   * (see CALCITE-2489).</p>
+   */
+  public static final CalciteSystemProperty<Boolean> ALLOW_FIELD_INDEX_ACCESS =
+      booleanProperty("calcite.experimental.allow.field.index.access", false);
```

Comment:
To make things more uniform with the properties above, I would rename it to: `calcite.enable.enumerable.fieldIndexAccess`. I added enumerable since it is strictly a property affecting the runtime not for parser, validator, etc.

Reply:
I agree on the property renaming, in light of the new name I have also moved it in the same group as the other "calcite.enable.enumerable.*" ones.